docs/latest/design/egctl.md

```diff
@@ -0,0 +1,61 @@
+# Introduce egctl
+
+## Motivation
+
+EG should provide a command line tool with following capabilities:
+
+- Collect configuration from envoy proxy and gateway
+- Update admin options such as changing log level
+- Analyse system configuration to diagnose any issues in envoy gateway
+
+This tool is named `egctl`.
+
+## Syntax
+
+Use the following syntax to run `egctl` commands from your terminal window:
+
+```console
+egctl [command] [entity] [name] [flags]
+```
+
+where `command`, `name`, and `flags` are:
+
+* `command`: Specifies the operation that you want to perform on one or more resources,
+  for example `config`, `version`.
+
+* `entity`: Specifies the entity the operation is being performed on such as `envoy-proxy` or `envoy-gateway`.
+
+* `name`: Specifies the name of the specified instance. 
+
+* `flags`: Specifies optional flags. For example, you can use the `-c` or `--config` flags to specify the values for installing.
+
+If you need help, run `egctl help` from the terminal window.
+
+## Operation
+
+The following table includes short descriptions and the general syntax for all the `egctl` operations:
+
+| Operation | Syntax                           | Description                                                                 |
+| --------- | -------------------------------- | --------------------------------------------------------------------------- |
+| `version` | `egctl version`                  | Prints out build version information.                                       |
+| `config`  | `egctl config ENTITY`            | Retrieve information about proxy configuration from envoy proxy and gateway |
+| `log`     | `egctl log ENTITY --level trace` | Change envoy proxy's log level                                              |
```

Comment:
Proxy configuration changes should be accomplished through a configuration manifest and not through an egctl subcommand. For example- `kubectl apply -f envoyproxy.yaml`. Xref https://github.com/envoyproxy/gateway/pull/842 for an example on configuring proxy logging through the EnvoyProxy API.

Reply:
will envoy proxy rolling after applying changes to `envoyproxy.yaml`? if so, the `log` will change proxy without a rolling.
Technically, it will port forward admin port and send a post request to [logging endpoint](https://www.envoyproxy.io/docs/envoy/latest/operations/admin#post--logging)

Reply:
The current implementation uses the `--log-level` CLI argument. Therefore managing logging verbosity through the `log` subcommand or the config file will cause a rolling restart without changing the implementation per https://www.envoyproxy.io/docs/envoy/latest/operations/admin#post--logging.